src/main/scala/cognite/spark/v1/AssetsRelation.scala

```diff
@@ -14,8 +14,10 @@ import org.apache.spark.sql.types._
 import org.apache.spark.sql.{Row, SQLContext}
 
 import java.time.Instant
+import scala.util.Try
 
-class AssetsRelation(config: RelationConfig)(val sqlContext: SQLContext)
+class AssetsRelation(config: RelationConfig, subtreeIds: Option[List[String]] = None)(
```

Comment:
my question here is: why do we take this from the options? We could add this to readSchema maybe and find in `sparkFilters`? I cannot think a nice way to reflect this in the UI... Because the options are most of the time write oriented in there 🤔

Reply:
The main reason is that it's not a data column, and it would give way to ugly mixes of filters (mainly with OR)
[Here ](https://cognitedata.atlassian.net/wiki/spaces/INT/pages/3721166999/Asset+subtrees+retrieval+on+transformations )are the different alternatives considered, what you describe would be Option 1, what is being implemented corresponds to Option 3.